### PR TITLE
Create `-sort-predicates` flag for eskip

### PR DESCRIPTION
**Pull request by MustafaSaber**
Signed-off-by: Mustafa Abdelrahman <mustafa.abdelrahman@zalando.de>

### CODE REVIEW

File: cmd/eskip/load.go [hunk 1]
```diff
@@ -126,7 +127,14 @@ func printCmd(a cmdArgs) error {
 			}
 		}
 
+		if sortPredicate {
+			for _, route := range lr.routes {
+				sort.Slice(route.Predicates, func(i, j int) bool { return i < j })
```
Comment (RomanZavodskikh) [inline cmd/eskip/load.go:132]: It seems to me such `sort.Slice` call does not actually mutate `route.Predicates` slice.
https://stackoverflow.com/questions/37695209/golang-sort-slice-ascending-or-descending 
As far as I see here to sort a slice with `sort.Slice` we need to use `i`, `j` variables as indexes to this slice. However, we compare here only indexes between each other without comparison *elements* of a slice.

Comment (RomanZavodskikh) [inline cmd/eskip/load.go:132]: Should there be some unit tests to check whether such sort actually works?

Comment (MustafaSaber) [reply cmd/eskip/load.go:132]: yes, that's in progress.

File: cmd/eskip/args.go [hunk 2]
```diff
@@ -39,6 +39,7 @@ const (
 	appendFiltersFlag  = "append"
 	appendFileFlag     = "append-file"
 	prettyFlag         = "pretty"
+	sortPredicateFlag  = "sort-predicates"
```
Comment (AlexanderYastrebov) [inline cmd/eskip/args.go:42]: Lest make it plural `sortPredicatesFlag`

File: eskip/string.go [hunk 3]
```diff
@@ -78,7 +80,7 @@ func argsString(args []interface{}) string {
 	return strings.Join(sargs, ", ")
 }
 
-func (r *Route) predicateString() string {
+func (r *Route) predicateString(prettyPrintInfo PrettyPrintInfo) string {
```
Comment (AlexanderYastrebov) [inline eskip/string.go:83]: We do not need whole PrettyPrintInfo here (it is also passed by value - creates a copy), lets just pass bool.

File: cmd/eskip/load.go (hunk 1 above)
Comment (MustafaSaber) [reply cmd/eskip/load.go]: Implemented in `eskip/string_test.go`

File: eskip/string.go [hunk 4]
```diff
@@ -107,9 +109,20 @@ func (r *Route) predicateString() string {
...
+		for _, p := range predicatesCopy {
+			if p.Name != "Any" {
+				predicates = appendFmt(predicates, "%s(%s)", p.Name, argsString(p.Args))
+			}
+		}
+	} else {
+		for _, p := range r.Predicates {
+			if p.Name != "Any" {
+				predicates = appendFmt(predicates, "%s(%s)", p.Name, argsString(p.Args))
+			}
```
Comment (AlexanderYastrebov) [inline eskip/string.go:125]: Lets have main logic loop outside of the condition something like:
```go
rp := r.Predicates
if sortPredicates {
    rp = make([]*Predicate, len(r.Predicates))
    copy(rp, r.Predicates)
    sort.Slice(rp, func(i, j int) bool { return rp[i].String() < rp[j].String() })
}

for _, p := range rp {
    if p.Name != "Any" {
        predicates = appendFmt(predicates, "%s(%s)", p.Name, argsString(p.Args))
    }
}
```

**Comment by MustafaSaber** [comment]
👍

**Pull request merged**